docusaurus/docs/React/custom-code-examples/livestream-setup.mdx

```diff
@@ -0,0 +1,77 @@
+
+The React SDK offers a special component for that purpose - the `VirtualizedMessageList`. The component renders only the messages visible in the viewport and dynamically updates its contents as the user scrolls. 
+This decreases browser memory usage and does not degrade the performance of the page even if the user receives thousands of messages in a single session. 
+
+In this article, we will go through a best practices and tweaks for configuring the virtualized message list. 
+
+## Get Started
+
+The `VirtualizedMessageList` is mostly compatible with the non-virtualized `MessageList`, although not all properties are the same. The simplest configuration and placement are the same:
+
+```tsx
+<Chat client={client}>
+  <ChannelList />
+  <Channel>
+    <VirtualizedMessageList />
+    <MessageInput />
+  </Channel>
+</Chat>
+```
+
+With this configuration, the virtualized list will look identical to the non-virtualized version. With the next few steps, we can configure the component for maximum performance.
+
+## Disable the automatic scroll to bottom transition
+
+By default, the virtualized message list will scroll down to display new messages using the `"smooth"` scroll behavior ([MDN link](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTo#examples)). This works well for moderately busy chat rooms but can be unwieldy in chats with more than 2-3 incoming messages per second. A safer, snappier option is `"auto"`. With that setting, the list scrolls to the latest received message instantly. 
+
+```tsx
+<VirtualizedMessageList stickToBottomScrollBehavior="auto" />
+```
+
+## Specify the default item height
+
+The virtualized message list supports items with varying heights and automatically keeps track of the rendered item sizes, while estimating the not-yet-rendered sizes based on latest message during load. To optimize this further and minimize potential recalculations, set the `defaultItemHeight` to the height of your usual one-line message (If in doubt, use the developer tools inspector to determine the sizing). 
+
+<img src={MessageHeight} alt='Message Height' width='700' />
+
+The element above is `58px` tall, so let's put that at the `defaultItemHeight` prop value:
+
+```tsx
+<VirtualizedMessageList defaultItemHeight={58} />
+```
+
+## Avoid vertical CSS margins
+
+This could be a potential pitfall if you're using a custom message component. The virtualized message list uses `getBoundingClientRect` ([MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect)) on the outermost message element. This method does not measure CSS margins, which are tricky to measure in the first place [due to collapsing](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Box_Model/Mastering_margin_collapsing)). If the custom message has vertical margins that affect its height, the message list might behave erratically (blinking, rendering white areas, etc). 
```

Comment:
Consider changing to the below, minor tweaks:
This could be a potential pitfall if you're using a custom `Message` component. The `VirtualizedMessageList` uses `getBoundingClientRect` ([MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect)) on the outermost message element. This method does not measure CSS margins, which are tricky to measure in the first place [due to collapsing](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Box_Model/Mastering_margin_collapsing)). If the custom `Message` component has vertical margins that affect its height, the message list might behave erratically (blinking, rendering white areas, etc).